### PR TITLE
IGNITE-22573 Fix warnings of IpFinderCleaner

### PR DESCRIPTION
**Pull request by J-Bakuli**
IpFinderCleaner periodically calls ServerImpl#pingNode(InetSocketAddress, UUID, UUID) [1] in order to ping addresses, registered in IpFinder. But, it causes excessive and useless warnings in server log [2], eg.:

```
[2024-06-14T12:26:33,978][WARN ][tcp-disco-ip-finder-cleaner-#5-#61][TcpDiscoverySpi] Failed to ping node [nodeId=null]. Reached the timeout 10000ms. Cause: Connection refused (Connection refused)
[2024-06-14T12:26:33,979][WARN ][tcp-disco-ip-finder-cleaner-#5-#61][TcpDiscoverySpi] Failed to ping node [nodeId=null]. Reached the timeout 10000ms. Cause: Connection refused (Connection refused)
[2024-06-14T12:26:33,979][WARN ][tcp-disco-ip-finder-cleaner-#5-#61][TcpDiscoverySpi] Failed to ping node [nodeId=null]. Reached the timeout 10000ms. Cause: Connection refused (Connection refused)
[2024-06-14T12:26:33,979][WARN ][tcp-disco-ip-finder-cleaner-#5-#61][TcpDiscoverySpi] Failed to ping node [nodeId=null]. Reached the timeout 10000ms. Cause: Connection refused (Connection refused)
[2024-06-14T12:26:33,980][WARN ][tcp-disco-ip-finder-cleaner-#5-#61][TcpDiscoverySpi] Failed to ping node [nodeId=null]. Reached the timeout 10000ms. Cause: Connection refused (Connection refused)
[2024-06-14T12:26:33,980][WARN ][tcp-disco-ip-finder-cleaner-#5-#61][TcpDiscoverySpi] Failed to ping node [nodeId=null]. Reached the timeout 10000ms. Cause: Connection refused (Connection refused)
[2024-06-14T12:26:33,981][WARN ][tcp-disco-ip-finder-cleaner-#5-#61][TcpDiscoverySpi] Failed to ping node [nodeId=null]. Reached the timeout 10000ms. Cause: Connection refused (Connection refused)
[2024-06-14T12:26:33,981][WARN ][tcp-disco-ip-finder-cleaner-#5-#61][TcpDiscoverySpi] Failed to ping node [nodeId=null]. Reached the timeout 10000ms. Cause: Connection refused (Connection refused)
[2024-06-14T12:26:33,981][WARN ][tcp-disco-ip-finder-cleaner-#5-#61][TcpDiscoverySpi] Failed to ping node [nodeId=null]. Reached the timeout 10000ms. Cause: Connection refused (Connection refused)
```

I have added address and changed WARN level to DEBUG for such output of IpFinderCleaner.

https://github.com/apache/ignite/blob/ac4319768bb6013aa401f5553d1b39de82e9751e/modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java#L2289
https://github.com/apache/ignite/blob/ac4319768bb6013aa401f5553d1b39de82e9751e/modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java#L936

[TCbot2](https://tcbot2.sbt-ignite-dev.ru/pr.html?serverId=apache&suiteId=IgniteTests24Java8_RunAll&branchForTc=pull/11637/head&action=Latest)
________________________________________________

Thank you for submitting the pull request to the Apache Ignite.

In order to streamline the review of the contribution 
we ask you to ensure the following steps have been taken:

### The Contribution Checklist
- [ ] There is a single JIRA ticket related to the pull request. 
- [ ] The web-link to the pull request is attached to the JIRA ticket.
- [ ] The JIRA ticket has the _Patch Available_ state.
- [ ] The pull request body describes changes that have been made. 
The description explains _WHAT_ and _WHY_ was made instead of _HOW_.
- [ ] The pull request title is treated as the final commit message. 
The following pattern must be used: `IGNITE-XXXX Change summary` where `XXXX` - number of JIRA issue.
- [ ] A reviewer has been mentioned through the JIRA comments 
(see [the Maintainers list](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute#HowtoContribute-ReviewProcessandMaintainers)) 
- [ ] The pull request has been checked by the Teamcity Bot and 
the `green visa` attached to the JIRA ticket (see [TC.Bot: Check PR](https://mtcga.gridgain.com/prs.html))

### Notes
- [How to Contribute](https://cwiki.apache.org/confluence/display/IGNITE/How+to+Contribute)
- [Coding abbreviation rules](https://cwiki.apache.org/confluence/display/IGNITE/Abbreviation+Rules)
- [Coding Guidelines](https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines)
- [Apache Ignite Teamcity Bot](https://cwiki.apache.org/confluence/display/IGNITE/Apache+Ignite+Teamcity+Bot)

If you need any help, please email dev@ignite.apache.org or ask anу advice on http://asf.slack.com _#ignite_ channel.

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 1]
```diff
@@ -937,8 +937,11 @@ private boolean pingNode(TcpDiscoveryNode node) {
 
                         if (IgniteSpiOperationTimeoutHelper.checkFailureTimeoutReached(e)
                             && (spi.failureDetectionTimeoutEnabled() || timeout != 0)) {
-                            log.warning("Failed to ping node [nodeId=" + nodeId + "]. Reached the timeout " +
-                                (timeout == 0 ? spi.failureDetectionTimeout() : timeout) + "ms. Cause: " + e.getMessage());
+                            if (log.isDebugEnabled()) {
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:940]: 1. pingNode can be called not only by cleanIpFinder, currently spams only cleanIpFinder method, but for other calls information about error is helpful and we shouldn't disable it.
2. Only timeout error is fixed, but there can be other causes. Next time pingNode can spam log again, but with another error message.
3. What about test?

Comment (J-Bakuli) [reply modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java]: @alex-plekhanov thank you, I agree with you and have made changes to the PR after your comments

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 2]
```diff
@@ -982,6 +987,43 @@ else if (!spi.failureDetectionTimeoutEnabled() && reconCnt == spi.getReconnectCo
...
+        catch (IgniteCheckedException e) {
+            logPingError(e.getMessage(), debugLogging);
+
+            throw e;
+        }
+    }
+
+    /** */
+    private void logPingError(String msg, boolean debugLogging) {
+        if (!debugLogging)
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:1021]: Let's invert logic and rename flag to `boolean logError`

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 3]
```diff
@@ -982,6 +987,43 @@ else if (!spi.failureDetectionTimeoutEnabled() && reconCnt == spi.getReconnectCo
...
+            boolean debugLogging) throws IgniteCheckedException {
+        try {
+            return pingNodeImpl(addr, nodeId, clientNodeId, timeout, debugLogging);
+        }
+        catch (IgniteCheckedException e) {
+            logPingError(e.getMessage(), debugLogging);
+
+            throw e;
+        }
+    }
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:1017]: Additional wrapper is redundant. Let's rename `pingNodeImpl` back to `pingNode` and use it with one additional parameter

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 4]
```diff
@@ -819,15 +819,17 @@ private boolean pingNode(TcpDiscoveryNode node) {
      * @param nodeId Node ID to ping. In case when client node ID is not null this node ID is an ID of the router node.
      * @param clientNodeId Client node ID.
      * @param timeout Timeout on operation in milliseconds. If 0, a value based on {@link TcpDiscoverySpi} is used.
+     * @param debugLogging Boolean flag indicating whether debug information should be printed into node log.
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:822]: Javadoc is confusing

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java [hunk 5]
```diff
@@ -2201,6 +2211,27 @@ public void testDuplicatedDiscoveryDataRemoved() throws Exception {
...
+        assertFalse(checkRestoreConnection());
+
+        setLoggerDebugLevel();
+
+        assertTrue(checkRestoreConnection());
+    }
+
+    /** @throws Exception If failed.*/
+    private boolean checkRestoreConnection() throws Exception {
+        ListeningTestLogger testLog = new ListeningTestLogger(log);
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java:2223]: Local variable is redundant, let's use testLog field.

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java [hunk 6]
```diff
@@ -2201,6 +2211,27 @@ public void testDuplicatedDiscoveryDataRemoved() throws Exception {
...
+
+        LogListener lsnr = LogListener
+                .matches(Pattern.compile("Failed to ping node \\[.*\\]\\. Reached the timeout"))
+                .build();
+
+        testLog.registerListener(lsnr);
+
+        boolean result;
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java:2231]: Local variable is redundant. `return lsnr.check()` can be used. But it's better to pass additional parameter for `checkRestoreConnection` and check it at the end of the method, something like: `assertEquals(logExpected, lsnr.check())`

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java [hunk 7]
```diff
@@ -2251,6 +2286,8 @@ public void testCheckRingLatency() throws Exception {
 
         testLog.registerListener(lsnr);
 
+        this.testLog = testLog;
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java:2289]: Local variable is redundant now

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java [hunk 8]
```diff
@@ -2251,6 +2286,8 @@ public void testCheckRingLatency() throws Exception {
...
+
         try {
             IgniteEx node = startGrid(getConfiguration("server").setGridLogger(testLog));
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java:2292]: `setGridLogger(testLog)` is redundant now

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java (hunk 7 above)
Comment (J-Bakuli) [reply modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java:2289]: thank you, agree

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java (hunk 3 above)
Comment (J-Bakuli) [reply modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:1017]: thank you, I agree

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java (hunk 4 above)
Comment (J-Bakuli) [reply modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:822]: changed to "@param debugLogging Debug flag

File: modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java [hunk 9]
```diff
@@ -2201,6 +2211,23 @@ public void testDuplicatedDiscoveryDataRemoved() throws Exception {
...
+        checkRestoreConnection(true);
+    }
+
+    /** @throws Exception If failed.*/
+    private void checkRestoreConnection(boolean logMsgExpected) throws Exception {
+        testLog = new ListeningTestLogger(log);
+
+        LogListener lsnr = LogListener
+                .matches(Pattern.compile("Failed to ping node \\[.*\\]\\. Reached the timeout"))
+                .build();
```
Comment (alex-plekhanov) [inline modules/core/src/test/java/org/apache/ignite/spi/discovery/tcp/TcpDiscoverySelfTest.java:2227]: Wrong indent

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 10]
```diff
@@ -7296,7 +7309,8 @@ private InetSocketAddress checkConnection(TcpDiscoveryNode node, int timeout) {
 
                             try {
                                 if (liveAddrHolder.get() == null) {
-                                    UUID id = pingNode(addr, node.id(), null, timeoutHelper.nextTimeoutChunk(perAddrTimeout)).get1();
+                                    UUID id = pingNode(addr, node.id(), null, timeoutHelper.nextTimeoutChunk(perAddrTimeout),
+                                            true).get1();
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:7313]: Wrong indent

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 11]
```diff
@@ -929,29 +933,30 @@ private boolean pingNode(TcpDiscoveryNode node) {
...
+                                    "Cause: " + e.getMessage(), logError);
 
                             break;
                         }
 
                         if (spi.isNodeStopping0()) {
-                            log.warning("Failed to ping node [nodeId=" + nodeId + "]. Current node is " +
-                                "stopping. Cause: " + e.getMessage());
+                            logPingError(errMsgPrefix + "Current node is stopping. " +
+                                    "Cause: " + e.getMessage(), logError);
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:959]: Wrong indent

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 12]
```diff
@@ -929,29 +933,30 @@ private boolean pingNode(TcpDiscoveryNode node) {
...
+                                    (timeout == 0 ? spi.failureDetectionTimeout() : timeout) +
+                                    "ms. Cause: " + e.getMessage(), logError);
 
                             break;
                         }
                         else if (!spi.failureDetectionTimeoutEnabled() && reconCnt == spi.getReconnectCount()) {
-                            log.warning("Failed to ping node [nodeId=" + nodeId + "]. Reached the reconnection " +
-                                "count " + spi.getReconnectCount() + ". Cause: " + e.getMessage());
+                            logPingError(errMsgPrefix + "Reached the reconnection count spi.getReconnectCount(). " +
+                                    "Cause: " + e.getMessage(), logError);
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:952]: Wrong indent

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 13]
```diff
@@ -929,29 +933,30 @@ private boolean pingNode(TcpDiscoveryNode node) {
...
                             break;
                         }
 
                         if (IgniteSpiOperationTimeoutHelper.checkFailureTimeoutReached(e)
                             && (spi.failureDetectionTimeoutEnabled() || timeout != 0)) {
-                            log.warning("Failed to ping node [nodeId=" + nodeId + "]. Reached the timeout " +
-                                (timeout == 0 ? spi.failureDetectionTimeout() : timeout) + "ms. Cause: " + e.getMessage());
+                            logPingError(errMsgPrefix + "Reached the timeout " +
+                                    (timeout == 0 ? spi.failureDetectionTimeout() : timeout) +
+                                    "ms. Cause: " + e.getMessage(), logError);
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:946]: Wrong indent

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 14]
```diff
@@ -929,29 +933,30 @@ private boolean pingNode(TcpDiscoveryNode node) {
                         reconCnt++;
 
                         if (!openedSock && reconCnt == 2) {
-                            log.warning("Failed to ping node [nodeId=" + nodeId + "]. Was unable to open the " +
-                                "socket at all. Cause: " + e.getMessage());
+                            logPingError(errMsgPrefix + "Was unable to open the socket at all. " +
+                                    "Cause: " + e.getMessage(), logError);
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:937]: Wrong indent

File: modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java [hunk 15]
```diff
@@ -912,9 +914,11 @@ private boolean pingNode(TcpDiscoveryNode node) {
                         return t;
                     }
                     catch (IOException | IgniteCheckedException e) {
+                        String errMsgPrefix = "Failed to ping node [nodeId=" + nodeId + ", address=" + addr + "]. ";
+
                         if (nodeId != null && !nodeAlive(nodeId)) {
-                            log.warning("Failed to ping node [nodeId=" + nodeId + "]. Node has left or is " +
-                                "leaving topology. Cause: " + e.getMessage());
+                            logPingError(errMsgPrefix + "Node has left or is leaving topology. " +
+                                    "Cause: " + e.getMessage(), logError);
```
Comment (alex-plekhanov) [inline modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java:921]: Wrong indent

**Pull request closed without merging**